manticore/ethereum/abi.py

```diff
@@ -137,6 +137,8 @@ def _serialize_bytes(value):
     def _serialize_tuple(types, value, dyn_offset=None):
         result = bytearray()
         dyn_result = bytearray()
+        if len(types) != len(value):
```

Comment:
We should probably use `if __debug__` in such cases as described here https://stackoverflow.com/a/27455446

Reply:
I've implemented the check the same way it's done in `_serialize_array`.

Personally I don't think it's the right choice for a tool like Manticore to trade the safety of an always-on check against the minuscule performance improvement from disabling an assert like this in release mode.

Reply:
imo i think @stephan-tolksdorf 's implementation is fine, and mirroring `_serialize_array` is correct. I actually think asserts and this explicit error checking are semantically different, and not necessarily interchangeable -- this case is more handling bad input, rather than expressing an invariant so I think the explicit check is more semantically correct. i don't think we should care too much about perf in this case. :)

Reply:
@mossberg Good point regarding the difference of checking preconditions on inputs vs checking internal invariants!